src/core/UBPreferencesController.cpp

```diff
@@ -829,27 +832,24 @@ void UBScreenListLineEdit::focusInEvent(QFocusEvent *focusEvent)
             button->setWindowFlag(Qt::Window, true);
             button->setWindowFlag(Qt::WindowDoesNotAcceptFocus, true);
             button->setText(QString::number(screenIndex++));
-            QFont font;
-            font.setPointSize(48);
             button->setFont(font);
             button->move(screen->geometry().topLeft());
             button->setMinimumSize(300, 150);
-            button->setDisabled(screenNames.contains(button->text()));
+            button->setDisabled(screenList.contains(button->text()));
             button->show();
 
             connect(button, &QPushButton::pressed, this, &UBScreenListLineEdit::addScreen);
 
             mScreenLabels << button;
         }
 
-        if (mValidator)
+        if (!mValidator)
         {
-            setValidator(nullptr);
-            delete mValidator;
+            mValidator = new UBStringListValidator(this);
+            setValidator(mValidator);
         }
 
-        mValidator = new UBStringListValidator(availableScreenIndexes);
-        setValidator(mValidator);
+        mValidator->setValidationStringList(availableScreenIndexes);
```

Comment:
`setValidScreenNames` ? Or `setAcceptableScreenNames` ? Or maybe even more accurate, `setAcceptableScreenIndexes`, with the idea to replace every mention of `screenNames` with `screenIndexes`